[quant][graphmode][fix] cloning schema in insert_observers

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #40653 [quant][graphmode][fix] remove unsupported ops in the list
* #40644 [quant] aten::repeat work for quantized tensor
* **#40624 [quant][graphmode][fix] cloning schema in insert_observers**

Summary:
Previously we didn't clone schema, so the default schema is used, this is
causing issue for some models

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22259519](https://our.internmc.facebook.com/intern/diff/D22259519)

jerryzh168 committed:
[quant][graphmode][fix] cloning schema in insert_observers

Summary:
Previously we didn't clone schema, so the default schema is used, this is
causing issue for some models

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] cloning schema in insert_observers"


Summary:
Previously we didn't clone schema, so the default schema is used, this is
causing issue for some models

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22259519](https://our.internmc.facebook.com/intern/diff/D22259519)

[ghstack-poisoned]

vkuzo reviewed (approved):
makes sense.  What was the error, just for own learning?

jerryzh168 commented:
> makes sense. What was the error, just for own learning?

forward function with default args failed to run after insert observers, I guess this info is probably in the schema.

btw, please take a look at the next PR as well, that is a fix for another model.

jerryzh168 committed:
Update on "[quant][graphmode][fix] cloning schema in insert_observers"


Summary:
Previously we didn't clone schema, so the default schema is used, this is
causing issue for some models

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22259519](https://our.internmc.facebook.com/intern/diff/D22259519)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] cloning schema in insert_observers"


Summary:
Previously we didn't clone schema, so the default schema is used, this is
causing issue for some models

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22259519](https://our.internmc.facebook.com/intern/diff/D22259519)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] cloning schema in insert_observers"


Summary:
Previously we didn't clone schema, so the default schema is used, this is
causing issue for some models

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22259519](https://our.internmc.facebook.com/intern/diff/D22259519)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] cloning schema in insert_observers"


Summary:
Previously we didn't clone schema, so the default schema is used, this is
causing issue for some models

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22259519](https://our.internmc.facebook.com/intern/diff/D22259519)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] cloning schema in insert_observers"


Summary:
Previously we didn't clone schema, so the default schema is used, this is
causing issue for some models

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22259519](https://our.internmc.facebook.com/intern/diff/D22259519)

[ghstack-poisoned]
